servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancerFactory.java

```diff
@@ -202,27 +211,59 @@ public RoundRobinLoadBalancerFactory.Builder<ResolvedAddress, C> healthCheckInte
          * <p>
          * {@link #healthCheckFailedConnectionsThreshold(int)} can be used to disable the health checking mechanism
          * and always consider all hosts for establishing new connections.
+         *
          * @param interval interval at which a background health check will be scheduled.
          * @param jitter the amount of jitter to apply to each retry {@code interval}.
          * @return {@code this}.
          * @see #healthCheckFailedConnectionsThreshold(int)
          */
         public RoundRobinLoadBalancerFactory.Builder<ResolvedAddress, C> healthCheckInterval(Duration interval,
                                                                                              Duration jitter) {
-            if (interval.isNegative() || interval.isZero()) {
-                throw new IllegalArgumentException("Health check interval should be greater than 0");
-            }
-            if (jitter.isNegative() || jitter.isZero()) {
-                throw new IllegalArgumentException("Jitter interval should be greater than 0");
-            }
-            if (interval.minus(jitter).isNegative() || interval.plus(jitter).isNegative()) {
-                throw new IllegalArgumentException("Jitter plus/minus interval underflow/overflow");
-            }
+            validate(interval, jitter);
             this.healthCheckInterval = interval;
             this.healthCheckJitter = jitter;
             return this;
         }
 
+        /**
+         * Configure an interval for re-subscribing to the original events stream in case all existing hosts become
```

Comment:
consider mentioning the default interval used if not overridden

Reply:
We made a decision not to leak any default values in javadoc everywhere in the project to give us more flexibility to change those in the future. We can reconsider the approach when we are 1.0.

Reply:
ah gotcha 👍